doc: add gauge apis (#1149)
lyft/envoy-mobile: docs/root/api/stats.rst

``StatsClient``
---------------

Envoy Mobile's stats function currently supports two type of stats: ``Counter`` and ``Gauge``.

> To use Envoy Mobile's stats function, obtain an instance of ``StatsClient`` from ``Engine`` (refer to :ref:`api_starting_envoy` for building an engine instance), and store the stats client to create ``Counter`` or ``Gauge`` instances. The following code examples show how to create a ``Counter``, and it's similar in the way a ``Gauge`` is created.

Reviewer: ```suggestion
To use Envoy Mobile's stats, obtain an instance of ``StatsClient`` from ``Engine`` (refer to :ref:`api_starting_envoy` for building an engine instance), and store the stats client to create ``Counter`` or ``Gauge`` instances:
```
I'd just add an example for gauges since that's what we do everywhere else, and it's nice to be comprehensive.
Also, please cap line widths as we do elsewhere in the file (100 characters IIRC)

Author: chatted shortly, keeping it as-is to avoid repetition. will cap the line.

Envoy Mobile's stats function currently supports two type of stats: ``Counter`` and ``Gauge``.

> To use Envoy Mobile's stats function, obtain an instance of ``StatsClient`` from ``Engine``

Reviewer: "function" should probably be "functionality"

(refer to :ref:`api_starting_envoy` for building an engine instance), and store the stats client to
create ``Counter`` or ``Gauge`` instances. The following code examples show how to create

> a ``Counter``, and it's similar in the way a ``Gauge`` is created.

Reviewer: ```suggestion
a ``Counter``, and the same approach is used to create a ``Gauge``.
```

  statsClient.counter(elements: ["foo", "bar"])

> The ``counter`` method from stats client takes a variable number of elements and returns a ``Counter`` instance, the elements are used to form a dot(.) delimited string, this string serves as the identifier of the counter. The string formed from the example code above is ``foo.bar``.

Reviewer: ```suggestion
The ``counter`` method from the stats client takes a variable number of elements and returns a ``Counter`` instance. The elements provided are used to form a dot(``.``) delimited string that serves as the identifier of the counter. The string formed from the example code above is ``foo.bar``.
```

> Store the counter instance, or the gauge instance if you are working with gauge.

Reviewer: ```suggestion
Store the instance of the counter or gauge, then use it to increment as necessary.
```

-----------
``Counter``
-----------

> A ``Counter`` can increment, call the ``increment`` method to increment the counter wherever it applies.

Reviewer: ```suggestion
A ``Counter`` can be incremented by calling the ``increment()`` method when applicable.
```

  // Increment the counter by 5
  // Swift
  counter.increment(count: 5)

---------
``Gauge``
---------

> A ``Gauge`` can be set with a value, added an amount to or subtracted an amount from.

Reviewer: ```suggestion

The value of a ``Gauge`` can be incremented, decremented, or reassigned.
```
